synapse/storage/event_federation.py

```diff
@@ -398,12 +398,11 @@ def _delete_old_forward_extrem_cache_txn(txn):
             sql = ("""
                 DELETE FROM stream_ordering_to_exterm
                 WHERE
-                (
-                    SELECT max(stream_ordering) AS stream_ordering
+                room_id IN (
+                    SELECT room_id AS stream_ordering
```

Comment:
(Oh, I was just blind and rich was pointing out the spurious `AS stream_ordering`)